ssas/service/admin/api_test.go

```diff
@@ -762,6 +788,39 @@ func (s *APITestSuite) TestRegisterDuplicateSystemIP() {
 	_ = ssas.CleanDatabase(group)
 }
 
+func (s *APITestSuite) TestRegisterSystemIPInvalidBody() {
+	group := ssas.Group{GroupID: "test-reset-creds-group"}
+	s.db.Create(&group)
+	system := ssas.System{GID: group.ID, ClientID: "test-reset-creds-client"}
+	s.db.Create(&system)
+
+	systemID := strconv.FormatUint(uint64(system.ID), 10)
+	body := `{"addr":"2.5.22.81"}`
+	req := httptest.NewRequest("POST", "/system/"+systemID+"/ip", strings.NewReader(body))
+	rctx := chi.NewRouteContext()
+	rctx.URLParams.Add("systemID", systemID)
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler := http.HandlerFunc(registerIP)
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusBadRequest, rr.Result().StatusCode)
+	_ = ssas.CleanDatabase(group)
```

Comment:
Go Testing package has some ability to automatically call setup and teardown functions. I don't think that the result of implementing it would benefit anything beyond cleaning up the code + setup, but wanted to share the docs on it in case you or @karinamzalez hadn't see it! https://pkg.go.dev/testing#hdr-Main